ocean_lib/web3_internal/contract_handler.py

```diff
@@ -52,13 +62,42 @@ def set_artifacts_path(artifacts_path):
 
     @staticmethod
     def _get(name, address=None):
+        """
+        Return the contract & its concise version, for a given name.
+
+        :param name: Contract name, str
+        :param address: hex str -- address of contract
+        :return: tuple of (contract, concise_contract)
+        """
         if address:
-            return ContractHandler._contracts.get(
-                (name, address)
-            ) or ContractHandler._load(name, address)
-        return ContractHandler._contracts.get(name) or ContractHandler._load(
-            name, address
-        )
+            tup = ContractHandler._contracts.get((name, address))
```

Comment:
I think you can set an input_tup as different, only. e.g.
input_tup = (name, address) if address else (name)

and reuse the same contents from the if/else only with a different input. It would reduce duplication.

Reply:
Good point. Fixed in 0a5976a